yjit/src/codegen.rs

```diff
@@ -3699,6 +3699,37 @@ fn gen_branchnil(
     EndBlock
 }
 
+fn gen_throw(
+    jit: &mut JITState,
+    ctx: &mut Context,
+    asm: &mut Assembler,
+    _ocb: &mut OutlinedCb,
+) -> CodegenStatus {
+    let throw_state = jit.get_arg(0).as_u64();
+    let throwobj = asm.load(ctx.stack_pop(1));
+
+    // THROW_DATA_NEW allocates. Save SP for GC and PC for allocation tracing as
+    // well as handling the catch table. However, not using jit_prepare_routine_call
+    // since we don't need a patch point for this implementation.
+    jit_save_pc(jit, asm);
+    gen_save_sp(asm, ctx);
+
+    extern "C" {
```

Comment:
Can you add a comment explaining a bit of what `rb_vm_throw()` does? What kind of work does it do before we exit to the interpreter?

Reply:
added a comment https://github.com/ruby/ruby/pull/7491/commits/c74bfe1e3789a7e451eac03595bb447ab0b0060b